samples/subsys/ipc/ipc_service/icmsg/src/main.c

```diff
@@ -97,48 +68,120 @@ int main(void)
 
 	ret = ipc_service_open_instance(ipc0_instance);
 	if ((ret < 0) && (ret != -EALREADY)) {
-		LOG_INF("ipc_service_open_instance() failure");
+		LOG_ERR("ipc_service_open_instance() failure");
 		return ret;
 	}
 
 	ret = ipc_service_register_endpoint(ipc0_instance, &ep, &ep_cfg);
 	if (ret < 0) {
-		printf("ipc_service_register_endpoint() failure");
+		LOG_ERR("ipc_service_register_endpoint() failure");
 		return ret;
 	}
 
 	k_sem_take(&bound_sem, K_FOREVER);
 
-	uint16_t mlen = MLEN_0;
+	LOG_INF("Perform sends for 1s");
+	int64_t start = k_uptime_get();
+	size_t mlen = PACKET_SIZE_START;
+	size_t bytes_sent = 0;
 
-	while (true) {
+	while ((k_uptime_get() - start) < 1000) {
 		ret = ipc_service_send(&ep, &msg, mlen);
 		if (ret == -ENOMEM) {
 			/* No space in the buffer. Retry. */
 			continue;
 		} else if (ret < 0) {
-			LOG_ERR("send_message(%d) failed with ret %d", msg.message, ret);
+			LOG_ERR("Failed to send (%c) failed with ret %d", msg.data[0], ret);
 			break;
 		}
 
-		msg.message++;
-		if (msg.message > 'z') {
-			msg.message = 'a';
+		msg.data[0]++;
+		if (msg.data[0] > 'z') {
+			msg.data[0] = 'a';
 		}
 
+		bytes_sent += mlen;
 		mlen++;
+
 		if (mlen > sizeof(struct data_packet)) {
-			mlen = MLEN_0;
+			mlen = PACKET_SIZE_START;
 		}
 
-		/* Quasi minimal busy wait time which allows to continuosly send
-		 * data without -ENOMEM error code. The purpose is to test max
-		 * throughput. Determined experimentally.
-		 */
-		k_busy_wait(100);
+		k_usleep(1);
 	}
 
-	LOG_INF("IPC-service HOST demo ended.");
+	LOG_INF("Sent %zu bytes", bytes_sent);
+
+	LOG_INF("Wait 500ms. Let net core finish its sends");
+	k_msleep(500);
+
+	LOG_INF("Stop network core");
+	nrf_reset_network_force_off(NRF_RESET, true);
+
+	LOG_INF("Reset IPC service");
+
+	ret = ipc_service_deregister_endpoint(&ep);
+	if (ret != 0) {
+		LOG_ERR("ipc_service_register_endpoint() failure");
+		return ret;
+	}
+
+	/* Reset message and expected message value and len. */
+	expected_message = 'A';
+	expected_len = PACKET_SIZE_START;
+	msg.data[0] = 'a';
+	mlen = PACKET_SIZE_START;
+
+	/* Reset bound sem. */
+	ret = k_sem_init(&bound_sem, 0, 1);
+	if (ret != 0) {
+		LOG_ERR("k_sem_init() failure");
+		return ret;
+	}
+
+	ret = ipc_service_register_endpoint(ipc0_instance, &ep, &ep_cfg);
+	if (ret != 0) {
+		LOG_INF("ipc_service_register_endpoint() failure");
+		return ret;
+	}
+
+	LOG_INF("Run network core");
+	nrf_reset_network_force_off(NRF_RESET, false);
+
+	k_sem_take(&bound_sem, K_FOREVER);
+
+	LOG_INF("Perform sends for 1s");
+	start = k_uptime_get();
+	bytes_sent = 0;
+
+	while ((k_uptime_get() - start) < 1000) {
```

Comment:
put in a function and use it here and at line 88

Reply:
ok